cmake/process_options.cmake

```diff
@@ -117,3 +123,68 @@ if(ENABLE_BLAS)
 else()
     message(STATUS "ENABLE_BLAS is OFF.")
 endif()
+
+if(ENABLE_THREADS)
+    message(STATUS "ENABLE_THREADS is ON.")
+    find_package(Threads)
+
+    if(NOT Threads_FOUND)
+        message(FATAL_ERROR "THREADS is enabled but not found.")
+    endif()
+
+    target_link_libraries(lightning_external_libs INTERFACE Threads::Threads)
+else()
+    message(STATUS "ENABLE_THREADS is OFF.")
+endif()
+
+if(ENABLE_KOKKOS)
+    if (NOT WIN32 AND NOT APPLE)
+        if(OpenMP_CXX_FOUND)
+            # Enable KOKKOS with the openmp backend.
+            option(Kokkos_ENABLE_OPENMP "Enable the Kokkos OPENMP device" ON)
+            message(STATUS "KOKKOS OPENMP DEVICE ENABLED.")
+        elseif(Threads_FOUND)
+            # Enable KOKKOS with the threads backend.
+            option(Kokkos_ENABLE_PTHREAD "Enable the Kokkos THREADS device" ON)
+            message(STATUS "KOKKOS THREADS DEVICE ENABLED.")
+        else()
+            # Enable KOKKOS with the serial backend.
+            option(Kokkos_ENABLE_SERIAL  "Enable Kokkos SERIAL device" ON)
+            message(STATUS "KOKKOS SERIAL DEVICE ENABLED.")
+        endif()
+    else()
+        # Enable KOKKOS with the serial backend.
+        option(Kokkos_ENABLE_SERIAL  "Enable Kokkos SERIAL device" ON)
+        message(STATUS "KOKKOS SERIAL DEVICE ENABLED--.")
+    endif()
+    # If no backend is specified, KOKKOS will enable the serial backend by default.
+
+    option(Kokkos_ENABLE_COMPLEX_ALIGN "Enable complex alignment in memory" OFF)
+
+    set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+    include(FetchContent)
+
+    FetchContent_Declare(kokkos
+                         GIT_REPOSITORY https://github.com/kokkos/kokkos.git
+                         GIT_TAG        3.5.00
```

Comment:
I assume the new release tag 3.6.00 does not improve things here?

Reply:
So, I checked it before, as soon as it was released, but I wanted to double-check because something could have changed.
It builds, but with a cascade of deprecated warnings from functions called internally by Kokkos.
We may find some improvements, but nothing evident for now.
I suggest keeping it as `3.5.00`, for now.

Reply:
I will change it to 3.6.00.